zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -679,6 +686,151 @@ benchmarks! {
+        //* assume at first that the last time frame is one block before the current time frame
+        let t = 0;
+        let current_time_frame: TimeFrame = last_offset + t.saturated_into::<u64>();
+        // for current_time_frame in last_offset..=(last_offset + t.saturated_into::<u64>()) {
+        for i in 1..=f {
+            <MarketIdsPerOpenTimeFrame<T>>::try_mutate(current_time_frame, |ids| {
+                // + 31 to not conflict with the markets of MarketIdsPerOpenBlock
+                ids.try_push((i + 31).into())
+            }).unwrap();
+        }
+        // }
+    }: {
+        Pallet::<T>::market_status_manager::<
+            _,
+            MarketIdsPerOpenBlock<T>,
+            MarketIdsPerOpenTimeFrame<T>
+            >(
+                block_number,
+                last_time_frame,
+                current_time_frame,
+                |market_id, market| {
+                    // noop, because weight is already measured somewhere else
+                    Ok(())
+                },
+            )
+            .unwrap();
+    }
+
+    market_resolution_manager {
+        let r in 1..31;
+        let d in 1..31;
+
+        let range_start: MomentOf<T> = 100_000u64.saturated_into();
+        let range_end: MomentOf<T> = 1_000_000u64.saturated_into();
+        // ensure markets exist
+        for _ in 0..64 {
+            let (_, market_id) = create_market_common::<T>(
+                MarketCreation::Permissionless,
+                MarketType::Categorical(T::MaxCategories::get()),
+                ScoringRule::CPMM,
+                Some(MarketPeriod::Timestamp(range_start..range_end)),
+            )?;
+            // ensure market is reported
+            T::MarketCommons::mutate_market(&market_id, |market| {
+                market.status = MarketStatus::Reported;
+                Ok(())
+            })?;
+        }
+
+        let block_number = 1u64.saturated_into::<T::BlockNumber>();
+
+        let mut r_ids_vec = Vec::new();
+        for i in 1..=r {
+           r_ids_vec.push(i.into());
+        }
+        MarketIdsPerReportBlock::<T>::mutate(block_number, |ids| {
+            *ids = BoundedVec::try_from(r_ids_vec).unwrap();
+        });
+
+        let mut d_ids_vec = Vec::new();
+        for i in 1..=d {
+            // + 31 to not conflict with the markets of MarketIdsPerReportBlock
+            d_ids_vec.push((i + 31).into());
+        }
+        MarketIdsPerDisputeBlock::<T>::mutate(block_number, |ids| {
+            *ids = BoundedVec::try_from(d_ids_vec).unwrap();
+        });
+
+        // TODO(#730) this will get outdated with the merged PR
+        let now: T::BlockNumber = T::DisputePeriod::get() + block_number;
+    }: {
+        Pallet::<T>::resolution_manager(
+            now,
+            |market_id, market| {
+                // noop, because weight is already measured somewhere else
+                Ok(())
+            },
+        ).unwrap();
+    }
+
+    on_initialize_top_overhead {
```

Comment:
So, what's happening here is that you're benchmarking the top level of the function by recreating its layout but deleting certain parts of it, right? I don't know man... this looks like a nightmare to maintain.

It should be easier to do this by using dependency injection. You inject the managers which open/close/resolve markets. For benchmarking the top level component, you then inject noop managers which take (virtually) no computing time.

Reply:
That's a good idea. Thanks a lot!

Reply:
3aec79b4a0b530726c1af595056f6c832135ddeb

I actually saw an unused `on_initialize_resolve_overhead` and just used it.